### DISCUSSION TITLE
Continue Function:

### DISCUSSION THREAD

**Discussion by spammenotinoz** [yokingma/search_with_ai#33] [General]
Excellent Project.  Thank-you.  With the default free providers is the "Continue to ask based on the current context" supported?
Or does this require either a LLM other than DuckDuckGo?

**Comment by spammenotinoz** [comment]
Bug in the FreeDuckDuckGo

**Reply by yokingma** [reply]
> Bug in the FreeDuckDuckGo

Thanks for liking this project, The FreeDuckDuckGo project is not stable and may not be available now. You'd better have your own KEY, such as OpenAI's key to call the GPT API:

Modify the env file:  .env.docker(or .env)
```env
# openai key
OPENAI_KEY=your_key
# openai proxy, default is for docker-compose, could modify if you need.
OPENAI_PROXY_URL=https://api.openai.com/v1
```

**Reply by spammenotinoz** [reply]
Yes indeed that's why I closed the issue.
Sorry for creating it.
I am using Claude 3 Haiku via LiteLLM proxy and boss it is highly impressive work.
I was already familiar with LiteLLM, and please do not take offence, when I debugged some of the client side code, I could see my API keys, this avoids exposing any real keys to the client side as I have placed on a Public Server (no authentication) but let friends and family be aware of the URL.

I didn't want to use your site and drive up your API costs!!
Thank-you really good work

**Comment by jajsfw** [comment]
自带的SEARXNG_HOSTNAME不能用

**Reply by yokingma** [reply]
> 自带的SEARXNG_HOSTNAME不能用

确保SEARXNG相关接口，服务器是能访问的吗？使用SEARXNG一般需要网络环境的支持。
